lib/syntax_tree/pattern.rb

```diff
@@ -123,7 +123,7 @@ def compile_node(root)
       elsif DynaSymbol === root and parts = root.parts and parts.size == 1 and TStringContent === parts[0]
         symbol = parts[0].value.to_sym
 
-        ->(attribute) { attribute == value }
+        ->(node) { node == symbol }
```

Comment:
I tried to add a test to validate this change but couldn't find a way to trigger this case.
I tried things like:
```ruby
    def test_search_string_simple
      results = search(%{"#{'abc'}"}, "StringLiteral[parts: [TStringContent[value: 'abc']]]")

      assert_equal 1, results.length
    end

    def test_search_symbol_simple
      puts %s{:"#{'abc'}"}
      puts ':"#{\'abc\'}"'
      results = search(':"#{\'abc\'}"', ':abc')
      results = search(':"#{\'abc\'}"', ':"#{\'abc\'}"')
      results = search(%{"#{'abc'}"}, "DynaSymbol[parts: [TStringContent[value: 'abc']]]")

      assert_equal 1, results.length
    end
```
But none seem to match this specific condition.